base/wrappers/data_source.go

```diff
@@ -48,6 +49,12 @@ func (s *dataSourceSyncFunction) SyncDataSource(ctx context.Context, config *dat
 		if err != nil {
 			logger.Error(fmt.Sprintf("Failure during data source sync: %v", err))
 		}
+
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic during access provider sync from target: %v", r)
```

Comment:
Message should be update per type of sync?

Reply:
Oh of course...